packages/flutter_test/lib/src/matchers.dart

```diff
@@ -452,11 +452,11 @@ Matcher coversSameAreaAs(Path expectedPath, { required Rect areaToCompare, int s
 ///    verify that two different code paths create identical images.
 ///  * [flutter_test] for a discussion of test configurations, whereby callers
 ///    may swap out the backend for this matcher.
-AsyncMatcher matchesGoldenFile(Object key, {int? version}) {
+AsyncMatcher matchesGoldenFile(Object key, { int? version, bool isFlaky = false }) {
```

Comment:
I am still not sure about exposing this to all Flutter developers... If a test fails, ideally it would be fixed, right? For our case, the cause of the failure is way more nuanced, a Flutter developer would not be digging into the depths of the engine code like this, and would not be able to to individually address their specific failure.

If a test fails, and the diff is acceptable to the user, they can just run `flutter test --update-goldens` and move on.

I don't know that the public API should have an opinion about flakiness.

Reply:
If the cause of flakiness is in the Flutter SDK, then sure. But flakiness can be caused by app code because:

- it depends on current date-time.
- it depends on `Math.random`.
- it contains a race condition due to asynchrony.
- it runs in non-uniform environment (different OS versions, different hardware).
- it has non-hermetic dependencies (e.g. pub dependencies shifting versions behind-the-scenes).

Having said that, "flakiness" specifically may be too specific. There may be other reasons a developer might want to label a test to pass but still generate a golden, such as:

- You see that a generated golden is completely non-sensical, but at the same time you don't know what it _should_ look like. So you don't want to update the golden to a non-sensical one, but you still want to generate it for human evaluation.
- You spot a test that's too brittle. It's trying to test the look of one button, but it's taking a golden of the entire screen (i.e. overtesting). It breaks too often, so you want to disable it, but alert the owner that this test's scope should be reduced.
- You want to divide your tests into two categories. One category is like a fully automated unit-test; it is expected to generate stable goldens and prevent code submissions that change them (this is the system Flutter uses). You also want a second category of tests that generate goldens but never fail. The system you submit the goldens to alerts humans (e.g. a QA team) to review any deltas and either approve or file issues.

So I think this option would be useful to developers. OTOH, all this is speculative. Do you have a recommendation for how to expose this flag to tests, if not through `matchesGoldenFile`?

Reply:
I think soliciting feedback from someone other than just me would be beneficial here. I think in all of these cases, the developer would want to just fix their test, not accept it as broken and have it fail silently.

Reply:
No, if these arguments do not immediately come across as sound, I'm totally fine with keeping it private.